datadotworld/client/api.py

```diff
@@ -53,12 +57,22 @@ def __init__(self, config):
             header_value='Bearer {}'.format(self._config.auth_token))
         swagger_client.user_agent = _user_agent()
 
+        content_negotiating_client = ContentNegotiatingApiClient(
+            host=self._host,
+            header_name='Authorization',
+            header_value='Bearer {}'.format(self._config.auth_token))
+        content_negotiating_client.user_agent = _user_agent()
+
+        # Create a base version of the content_negotiating_client object
+        self._sql_client = copy.copy(content_negotiating_client)
```

Comment:
This is better, but potentially unsafe, because:
1. Another developer may choose to reuse one of these instance variables in other functions and encounter non-deterministic behavior because `sql()`, `sparql()` and `append_stream()` modify them.
2. In a multi-thread context, a client may call `sql()` indicating the desired mime type and get something else in return, because of a race condition.

Finally, functions with side-effects should be avoided if possible and, here, I think they could be avoided.

Please consider one final change before you merge.
Instead of `self._sql_client = copy.copy(content_negotiating_client)`, how about:
```python
self._build_api_client = functools.partial(
    ContentNegotiatingApiClient, 
    host=self._host, 
    header_name='Authorization', 
    header_value='Bearer {}'.format(self._config.auth_token))
```

That way, in `sql()` and `sparql()` you can create instances on demand:
```python
api_client = self._build_api_client(default_mimetype_header_accept=desired_mimetype)
```
And, in `append_stream()`:
```python
api_client = self._build_api_client(default_mimetype_header_content_type=provided_mimetype)

Reply:
👍